corehq/apps/reports/migrations/0020_tableauserver_get_reports_using_role.py

```diff
@@ -0,0 +1,42 @@
+# Generated by Django 4.2.14 on 2024-08-13 19:14
+
+from django.db import migrations, models
+from corehq.util.django_migrations import skip_on_fresh_install
+
+
+@skip_on_fresh_install
+def _move_ff_to_setting(apps, schema_editor):
+    try:
+        from corehq.toggles import EMBED_TABLEAU_REPORT_BY_USER
+        from corehq.toggles import EMBEDDED_TABLEAU
+    except ImportError:
+        return
+    TableauServer = apps.get_model('reports', 'TableauServer')
+    for domain in EMBEDDED_TABLEAU.get_enabled_domains():
+        if not EMBED_TABLEAU_REPORT_BY_USER.enabled(domain):
+            try:
+                server = TableauServer.objects.get(domain=domain)
+            except TableauServer.DoesNotExist:
+                pass
+            server.get_reports_using_role = True
+            server.save()
```

Comment:
```suggestion
            except TableauServer.DoesNotExist:
                pass
            else:
                server.get_reports_using_role = True
                server.save()
```
As-written, if the exception is caught, `server` will be the server from a previous iteration.  I think you could also use `next` instead of `pass`, to go to the next iteration of the loop, but that always makes me nervous for some reason.

Reply:
also if you're feeling fancy, you could do:
```python
domains_using_role = (set(EMBEDDED_TABLEAU.get_enabled_domains())
                      - set(EMBEDDED_TABLEAU_REPORT_BY_USER.get_enabled_domains()))
TableauServer.objects.filter(domain__in=domains_using_role).update(get_reports_using_role=True)
```
Though in this context I'd weight clarity and safety far higher than performance, so whichever you find more intuitive.

Reply:
Yeah, it's a bit weird, but that's how it works:
```python
>>> def square(n):
...     if n > 4:
...         raise Exception("too complicated")
...     return n**2 
>>> for i in range(8):
...     try:
...         a = square(i)
...     except Exception:
...         pass
...     print(a)
0
1
4
9
16
16
16
16
```
This is sometimes useful for stuff like this:
```python
def greatest_divisor(n):
    for i in range(1, n):
        if n % i == 0:
            biggest = i
    print(f"The greatest divisor is {biggest}")
```
In older versions of python, loop variables from list comprehensions weren't local!